src/main/scala/is/hail/sparkextras/RepartitionedOrderedRDD2.scala

```diff
@@ -13,21 +13,30 @@ import org.apache.spark.sql.Row
   * needed. No assumption should need to be made about partition keys, but currently
   * assumes old partition key type is a prefix of the new partition key type.
   */
+object RepartitionedOrderedRDD2 {
+  def apply(prev: OrderedRVD, newPartitioner: OrderedRVDPartitioner): RepartitionedOrderedRDD2 = {
+    val dep = new OrderedDependency(prev.partitioner, newPartitioner, prev.rdd)
+    new RepartitionedOrderedRDD2(prev.rdd, prev.typ, dep, newPartitioner)
+  }
+}
+
 class RepartitionedOrderedRDD2(
-    prev: OrderedRVD,
+    prevRDD: RDD[RegionValue],
+    typ: OrderedRVDType,
+    dependency: OrderedDependency,
     newPartitioner: OrderedRVDPartitioner)
```

Comment:
There's some redundancy here that might be asking for trouble. Maybe have the constructor arguments be
```scala
prevRDD: RDD[RegionValue],
typ: OrderedRVDType,
oldPartitioner: OrderedRVDPartitioner,
newPartitioner: OrderedRVDPartitioner
```
which is enough to define the `OrderedDependency`? Also, that way users of `RepartitionedOrderedRDD2` (or whatever we call it) don't need to know the implementation details of `OrderedDependency`.

Reply:
ok, thanks!